File: src/main/java/io/mfedirko/common/infra/dynamodb/DynamoLessonMapper.java
package io.mfedirko.common.infra.dynamodb;

import com.github.rjeschke.txtmark.Processor;
import io.mfedirko.common.util.DateHelper;
import io.mfedirko.learning.Lesson;
import org.springframework.stereotype.Component;

@Component
public class DynamoLessonMapper {
    public Lesson toLesson(DynamoLesson dynamoLesson) {
        String rawDescription = dynamoLesson.getDescription();
        String mdParsedDescription = Processor.process(rawDescription);
        return Lesson.builder()
                .parsedDescription(mdParsedDescription)
                .description(rawDescription)
                .title(dynamoLesson.getTitle())
                .category(dynamoLesson.getCategory())
                .author(dynamoLesson.getAuthor())
                .creationTimestampMillis(dynamoLesson.getCreationTimestampMillis())
                .creationTimestamp(DateHelper.unixMillisToLocalDateTime(dynamoLesson.getCreationTimestampMillis()))
                .build();
    }
}

File: src/main/kotlin/io/mfedirko/common/infra/dynamodb/DynamoLessonMapper.kt
package io.mfedirko.common.infra.dynamodb

import com.github.rjeschke.txtmark.Processor
import io.mfedirko.common.util.DateHelper
import io.mfedirko.learning.Lesson
import org.springframework.stereotype.Component

@Component
class DynamoLessonMapper {
    fun toLesson(dynamoLesson: DynamoLesson): Lesson {
        val rawDescription = dynamoLesson.description
        val mdParsedDescription = Processor.process(rawDescription)
        return Lesson().apply {
            parsedDescription = mdParsedDescription
            description = rawDescription!!
            title = dynamoLesson.title!!
            category = dynamoLesson.category!!
            author = dynamoLesson.author!!
            creationTimestampMillis = dynamoLesson.creationTimestampMillis!!
            creationTimestamp = DateHelper.unixMillisToLocalDateTime(dynamoLesson.creationTimestampMillis!!)
        }
    }
}